openfl/federated/data/sources/s3_data_source.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import hashlib
+import os
+from typing import Generator
+from urllib.parse import urlparse
+
+import boto3
+
+from openfl.federated.data.sources.data_source import DataSource, DataSourceType
+
+
+class S3DataSource(DataSource):
+    """Class for S3 data"""
+
+    def __init__(
+        self,
+        uri: str,
+        endpoint=None,
+        access_key_env_name=None,
+        secret_key_env_name=None,
+        hash_func=None,
+    ):
+        super().__init__(DataSourceType.S3)
+        self.uri = uri
+        self.endpoint = endpoint
+        self.access_key_env_name = access_key_env_name
```

Comment:
I understand now your config file.
In the S3DataSource there shall be the access_key and secret_key and not the env name.
Can you rename the vars to `access_key` and `secret_key` this is simpler. thx